### PR TITLE
Add expander component

### LINKED ISSUE #159: [Change Request]: Add Expander component
### What type of enhancement/feature is this?

New UI Component, Documentation Improvements

### Description

Add a Expander component to the Omni-components library and be aware of the following requirements

- Introduce a parent component to slot multiple Expander components ()
- Introduce the ability to slot content into the header of the Expander.  (Prefix, header/label and suffix slots).
- Add ability to set a label/header for the component.
- Add required custom events.
- Add relevant tests and documentation 

### Use Case/Reasoning for enhancement/feature.

Required component for the next version.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct
- [X] I have [searched for existing issues](https://github.com/capitec/omni-components/issues) and to the best of my knowledge this is not a duplicate
- [X] This is a single enhancement/feature

### PR DESCRIPTION
**Pull request by chromaticWaster**
### Description:

 **Add Expander and Expander Group components**

closes #159 

- Added the Expander and Expander Group components.
- Added visual and behaviour tests.
- Added relevant stories.

(*Link to bug/feature from [Issues](../issues) and how you resolve it*)

### Screenshots (if appropriate):

**Expander**

https://github.com/capitec/omni-components/assets/22874454/6971e9ae-49e5-4016-8d3f-0d3ea5dfc698

**Expander Group** 

https://github.com/capitec/omni-components/assets/22874454/be63f664-3869-4dcf-980d-9c6ee66104e4

**Expander test results** 

image

**Expander Group test results**
 
image

### All Submissions:

* [x] I have followed the [contribution guidelines](https://github.com/capitec/omni-components/blob/develop/CONTRIBUTING.md) for this project.
* [x] I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change.

### Changes to Core Features:

* [x] I have added an explanation of what my changes do and why I would like to include them.
* [x] I have written new tests for my core changes, as applicable.
* [x] I have successfully ran tests with my changes locally.
* [x] I have added/updated docs, as applicable.

### CODE REVIEW

**Review by stefan505 — APPROVED** [review]

File: src/expander/Expander.ts [hunk 1]
```diff
@@ -0,0 +1,459 @@
...
+ * <omni-expander label="My Expander">
+ *  <omni-label label="Expanded"></omni-label>
+ * </omni-expander>
+ * ```
+ * @element omni-expander
+ *
+ * @slot - Content to render inside the expander once expanded.
+ * @slot expand-icon - Replaces the expand icon by default this will be the omni-chevron-down-icon.
+ * @slot header-icon - Replaces the icon in the header which is usually placed on the opposite end of the expand icon.
+ *
```
Comment (BOTLANNER) [inline src/expander/Expander.ts:25]: Document events

File: src/expander/ExpanderGroup.ts [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+     * @attr [expand-mode]
+     */
+    @property({ type: String, reflect: true, attribute: 'expand-mode' }) expandMode?: 'multiple' | 'single' = 'single';
+
+    @state() private _observer: MutationObserver | undefined;
+
+    override connectedCallback(): void {
+        super.connectedCallback();
+
+        this._observer = new MutationObserver((mutations) => {
```
Comment (BOTLANNER) [inline src/expander/ExpanderGroup.ts:46]: Consider using events instead

**Review by BOTLANNER — CHANGES_REQUESTED** [review]

File: src/expander/Expander.ts (hunk 1 above)
Comment (chromaticWaster) [reply src/expander/Expander.ts:25]: Documented as requested.

File: src/expander/ExpanderGroup.ts (hunk 2 above)
Comment (chromaticWaster) [reply src/expander/ExpanderGroup.ts]: Updated to use events as requested.

**Review by BOTLANNER — APPROVED** [review]

**Pull request merged**